baseband/vdif/frame.py

```diff
@@ -395,8 +404,17 @@ def _get_frames(self, item):
     # Let frameset behave appropriately.
     def __getitem__(self, item=()):
         if isinstance(item, six.string_types):
-            # Header behaves as a dictionary.
-            return self.header0.__getitem__(item)
+            # Header behaves as a dictionary.  Assume base keywords are the
+            # same for every frame, except for thread_id (must be different)
+            # and invalid_data (can be different).
+            if item == 'thread_id':
+                return np.array([f.header[item] for f in self.frames])
```

Comment:
Possibly silly, but the collection of thread_ids are generally passed around as a list, so maybe we don't need to convert it to an array here?

Reply:
Oh, I just saw what you do below.  Never mind!

Reply:
Not so silly, I had it as a list initially, and could always return a list. The only reason I decided not to, is that at the receiving end it is much easier to check whether something is a scalar or a list now (from `shape` or `ndim`).